apps/rush-lib/src/logic/ProjectWatcher.ts

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { FSWatcher } from 'chokidar';
+import { PackageChangeAnalyzer } from './PackageChangeAnalyzer';
+import { RushConfiguration } from '../api/RushConfiguration';
+import { RushConfigurationProject } from '../api/RushConfigurationProject';
+import { Path } from '@rushstack/node-core-library';
+
+export interface IProjectWatcherOptions {
+  debounceMilliseconds?: number;
+  rushConfiguration: RushConfiguration;
+  selection: ReadonlySet<RushConfigurationProject>;
```

Comment:
```suggestion
  projectsToMonitor: ReadonlySet<RushConfigurationProject>;
```

Reply:
went with `projectsToWatch` for consistency